enhancements/olm/runtime-bundle-configuration.md

```diff
@@ -0,0 +1,796 @@
+7. user rectifies configuration error
+8. OLM detects configuration changes
+9. ClusterExtension conditions report success and content is installed/updated
+
+Bundle schema validation can fail for several reasons including:
+- configuration was provided by the user, but the resolved bundle does not provide a configuration schema
+- configuration includes keys that do not exist in the bundle configuration schema
+- configuration does not set a value for a required field
+- configuration sets a value of the wrong type (e.g. string where boolean is expected)
+- configuration sets a value outside the allowable range (e.g. value must be <= 10 but got 11) or enum set (must be one of \["Turtle", "Elephant", "Capybara"\] but got "Lion")
+- configuration sets a value that does not respect the desired format (e.g. value must be a valid url)
+
+Alternatively, the user can address the issue by returning to the previous happy state by rolling back to a previous version using the
+rollback flow described at the bottom of this section.
+
+#### 2. Upgrade Across Non-Detectable Breaking Changes
+
+Some breaking API changes such as changes in default behavior, or semantic meaning of fields, can be hard, or impossible,
+to detect and _could_ potentially have a large blast radius. By their nature, they are hard to detect. Such an issue would
+represent a bug for the content author to address. If the user detects an issue and suspects this to be the case, they can
+follow the rollback flow described at the bottom of this section.
+
+#### 3. Installing an Operator in SingleNamespace or OwnNamespace Modes
+
+1. user searches catalog for content
+2. user reads documentation and understands `watchNamespace` can be used to specify a namespace the operator should watch and whether it can or not be the install namespace
+3. user creates a ClusterExtension CR to install that content specifying `.spec.config.inline.watchNamespace` to the desired namespace
+4. OLM resolves the bundle
+5. The manifest provisioning layer gets the bundle configuration schema and validates the configuration:
+    1. `watchNamespace` is a valid namespace name (follows the right format)
+    2. `watchNamespace` is not the install namespace if the bundle does not support OwnNamespace
+    3. `watchNamespace` is the install namespace if the bundle does not support SingleNamespace
+6. OLM applies generated manifests to cluster
+7. ClusterExtension conditions report success
+
+#### Rollback Flow
+
+First, one must determine the version to rollback to.
+If the ClusterExtension attempted an upgrade and is in a failed state, `.status.install` should report the version
+of the installed bundle. If the current version was successfully installed in a way that violates the user intent, e.g.
+there was a non-detectable breaking change in configuration behavior between bundle versions, then, at present, one needs
+to consult the catalog and walk the upgrade graph backwards from the currently installed version.
```

Comment:
Is there any tooling/documentation that makes it easy for an end-user to reverse-traverse this upgrade graph from the currently installed version?